components/renku_data_services/data_connectors/db.py

```diff
@@ -297,6 +300,160 @@ async def delete_data_connector(
+
+        data_connector = (
+            await session.scalars(
+                select(schemas.DataConnectorORM).where(schemas.DataConnectorORM.id == link.data_connector_id)
+            )
+        ).one_or_none()
+        if data_connector is None:
+            raise errors.MissingResourceError(
+                message=f"Data connector with id '{link.data_connector_id}' does not exist or you do not have access to it."  # noqa E501
+            )
+
+        project = (
+            await session.scalars(select(schemas.ProjectORM).where(schemas.ProjectORM.id == link.project_id))
+        ).one_or_none()
+        if project is None:
+            raise errors.MissingResourceError(
+                message=f"Project with id '{link.project_id}' does not exist or you do not have access to it."
+            )
+
+        existing_link = await session.scalar(
+            select(schemas.DataConnectorToProjectLinkORM)
+            .where(schemas.DataConnectorToProjectLinkORM.data_connector_id == link.data_connector_id)
+            .where(schemas.DataConnectorToProjectLinkORM.project_id == link.project_id)
+        )
+        if existing_link is not None:
+            raise errors.ConflictError(
+                message=f"A link from data connector {link.data_connector_id} to project {link.project_id} already exists."  # noqa E501
+            )
+
+        link_orm = schemas.DataConnectorToProjectLinkORM(
+            data_connector_id=link.data_connector_id,
+            project_id=link.project_id,
+            created_by_id=user.id,
+        )
+
+        session.add(link_orm)
+        await session.flush()
+        await session.refresh(link_orm)
+
+        return link_orm.dump()
+
+    @with_db_transaction
+    @Authz.authz_change(AuthzOperation.delete_link, ResourceType.data_connector)
+    async def delete_link(
+        self,
+        user: base_models.APIUser,
+        data_connector_id: ULID,
+        link_id: ULID,
+        *,
+        session: AsyncSession | None = None,
+    ) -> models.DataConnectorToProjectLink | None:
+        """Delete a link from a data connector to a project."""
+        if not session:
+            raise errors.ProgrammingError(message="A database session is required.")
+
+        link_orm = (
+            await session.scalars(
+                select(schemas.DataConnectorToProjectLinkORM)
+                .where(schemas.DataConnectorToProjectLinkORM.id == link_id)
+                .where(schemas.DataConnectorToProjectLinkORM.data_connector_id == data_connector_id)
+            )
+        ).one_or_none()
+        if link_orm is None:
+            return None
+
+        authorized_from = await self.authz.has_permission(
+            user, ResourceType.data_connector, data_connector_id, Scope.DELETE
+        )
+        authorized_to = await self.authz.has_permission(user, ResourceType.project, link_orm.project_id, Scope.WRITE)
+        authorized = authorized_from or authorized_to
+        if not authorized:
```

Comment:
These are already checked in `authz.py` and should be removed from here.

Reply:
Done in 968df0f89befff1c2e5564770fdccd0638479432.